rusk-wallet/src/wallet.rs

```diff
@@ -1151,6 +1151,15 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
         }
     }
 
+    /// Check if the wallet is synced
+    pub async fn is_synced(&mut self) -> Result<bool, Error> {
+        let state = self.state()?;
+        let db_pos = state.cache().last_pos()?.unwrap_or(0);
+        let network_last_pos = state.fetch_network_pos().await? - 1;
```

Comment:
This only works because our tree is append only. But if we change that to be a sparse tree in the future, the logic makes no sense anymore.
Better to check for the total amount of notes that are synced and not for the position of the last note that had been synced.

Reply:
We cannot check for total amount of notes because we only store notes that we own